habitat-lab/habitat/datasets/rearrange/run_episode_generator.py

```diff
@@ -26,77 +26,70 @@
     from habitat.config import DictConfig
 
 
-@dataclass
+@attrs.define(auto_attribs=True)
 class SceneSamplerParamsConfig:
     scene: str = "v3_sc1_staging_00"
-    scene_sets: List[Any] = field(default_factory=list)
+    scene_sets: List[Any] = []
```

Comment:
I am against doing this as this technically bad behavior. I would rather we stick to dataclasses and wrap it with this: https://pypi.org/project/dataslots/

Reply:
The main issue here is that the list can be edited and it will affect the DEFAULT parameters of the config which is unexpected behavior and bugprone.